src/foam/mlang/mlang.js

```diff
@@ -457,6 +486,27 @@ foam.CLASS({
       ],
       javaCode: '//noop',
       swiftCode: '//noop'
+    },
+    {
+      name: 'authorize',
+      javaCode:`
+  return true;
+  `
+    },
+    {
+      name:'authorizeArg',
```

Comment:
The PropertyInfo's implement the Predicate interface, which means that they also have the authorize() method. Move this logic to the PropertyInfo itself and the design will be simpler.

Reply:
It seems like PropertyInfo extends SQLStatement interface, Predicate also extends SQLStatement interface. I don't see any inheritance between PropertyInfo and Predicate. Correct me if I am misunderstanding something.